api/src/test/java/org/openmrs/ObsTest.java

```diff
@@ -331,6 +331,44 @@ public void setValueAsString_shouldFailIfTheValueOfTheStringIsNull() throws Exce
 		Obs obs = new Obs();
 		obs.setValueAsString(null);
 	}
+
+	/**
+	 * @see Obs#setValueAsString(String)
+	 */
+	@Test
+	public void setValueAsString_shouldSetValueAsDateTimeIfDataTypeOfQuestionConceptIsDateTimeHavingTimeZone()
+		throws Exception {
+		Obs obs = new Obs();
+		ConceptDatatype cdt = new ConceptDatatype();
+		Concept c = new Concept();
+		c.setDatatype(cdt);
+		obs.setConcept(c);
+
+		String[] dates1 = { "2016-01-12T06:00:00+05:30", "2016-01-12T06:00:00+0530" };
+		String[] dates2 = { "2014-02-20T11:00:00.000-05:00", "2014-02-20T11:00:00.000-05" };
```

Comment:
What is the point of splitting the dates in 2 arrays ?

Reply:
The second array was meant to provide dates that do not conform to any of the formats. I have made this clear in the recent commit i just made.